internal/provider/apikey_resource.go

```diff
@@ -0,0 +1,325 @@
+		Description types.String   `tfsdk:"description"`
+		ExpiryTime  types.String   `tfsdk:"expiry_time"` // ISO 8601 format
+		Timeouts    timeouts.Value `tfsdk:"timeouts"`
+	}
+)
+
+var (
+	_ resource.Resource                = (*apiKeyResource)(nil)
+	_ resource.ResourceWithConfigure   = (*apiKeyResource)(nil)
+	_ resource.ResourceWithImportState = (*apiKeyResource)(nil)
+)
+
+func NewApiKeyResource() resource.Resource {
+	return &apiKeyResource{}
+}
+
+func (r *apiKeyResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*client.Client)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected *client.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = client
+}
+
+func (r *apiKeyResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_apikey"
+}
+
+func (r *apiKeyResource) Schema(ctx context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Provisions a Temporal Cloud API key.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "The unique identifier of the API key.",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"state": schema.StringAttribute{
+				Description: "The current state of the API key.",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"owner_type": schema.StringAttribute{
+				Description: "The type of the owner to create the API key.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"owner_id": schema.StringAttribute{
+				Description: "The ID of the owner to create the API key for.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"display_name": schema.StringAttribute{
+				Description: "The display name for the API key.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"description": schema.StringAttribute{
+				Description: "The description for the API key.",
+				Optional:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"expiry_time": schema.StringAttribute{
+				Description: "The expiry time for the API key in ISO 8601 format.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+		},
```

Comment:
please add [disabled](https://github.com/temporalio/api-cloud/blob/7887a9cf05d64dd598dc0054f99b815ef46e9517/temporal/api/cloud/identity/v1/message.proto#L182) to the definition to enable/disable the key via TF

Reply:
will add!